src/ZkFold/Symbolic/Data/ByteString.hs

```diff
@@ -355,3 +357,11 @@ instance (Symbolic c, NumberOfBits (BaseField c) ~ n) => Iso (FieldElement c) (B
 
 instance (Symbolic c, NumberOfBits (BaseField c) ~ n) => Iso (ByteString n c) (FieldElement c) where
   from (ByteString a) = fromBinary a
+
+instance (FromConstant Natural (ByteString 8 c), Concat (ByteString 8 c) (ByteString n c))
```

Comment:
Much better, but now we still need some tests to make sure this is right
I'd propose to add a test to `Tests/ByteString.hs` module which checks that `parseJSON . toJSON = Just`

Reply:
Please also add same JSON property test but for the `UInt`s too. These are good tests to have for these basic symbolic types.